framework/src/org/checkerframework/common/value/Range.java

```diff
@@ -0,0 +1,417 @@
+     * @return the refined {@code Range}.
+     */
+    public Range greaterThan(Range right) {
+        return new Range(Math.max(from, right.from + 1), to);
+    }
+
+    /**
+     * Determines a refined range of a variable that fall within this {@code Range} under the
+     * condition that this variable is greater than or equal to another variable that fall within
+     * the specified {@code Range}. This is used for calculating the control flow refined result of
+     * the &gt;= operator. i.e.
+     *
+     * <pre>
+     * <code>
+     *     {@literal @}IntRange(from = 0, to = 10) int a;
+     *     {@literal @}IntRange(from = 3, to = 5) int b;
+     *     ...
+     *     if (a &gt;= b) {
+     *         // range of <i>a</i> is now refined to [5, 10] because a value in range [0, 4]
+     *         // cannot be greater than or equal to variable <i>b</i> with range [3, 5]
+     *         ...
+     *     }
+     * </code>
+     * </pre>
+     *
+     * @param right the specified {@code Range} to compare with.
+     * @return the refined {@code Range}.
+     */
+    public Range greaterThanEq(Range right) {
+        return new Range(Math.max(from, right.from), to);
+    }
+
+    /**
+     * Determines a refined range of a variable that fall within this {@code Range} under the
+     * condition that this variable is equal to another variable that fall within the specified
+     * {@code Range}. This is used for calculating the control flow refined result of the ==
+     * operator. i.e.
+     *
+     * <pre>
+     * <code>
+     *     {@literal @}IntRange(from = 0, to = 10) int a;
+     *     {@literal @}IntRange(from = 3, to = 15) int b;
+     *     ...
+     *     if (a == b) {
+     *         // range of <i>a</i> is now refined to [3. 10] because a value in range [0, 2]
+     *         // cannot be equal to variable <i>b</i> with range [3, 15]
+     *         ...
+     *     }
+     * </code>
+     * </pre>
+     *
+     * @param right the specified {@code Range} to compare with.
+     * @return the refined {@code Range}.
+     */
+    public Range equalTo(Range right) {
+        return new Range(Math.max(from, right.from), Math.min(to, right.to));
+    }
+
+    /**
+     * Returns the number of possible values enclosed by this range. To prevent overflow, we use
+     * BigInteger for calculation and return a BitInteger.
+     *
+     * @return the number of possible values enclosed by this range
+     */
+    public BigInteger numberOfPossibleValues() {
+        return BigInteger.valueOf(to).subtract(BigInteger.valueOf(from)).add(BigInteger.valueOf(1));
+    }
+
+    /**
+     * Determines if the range is wider than a given value, i.e., if the number of possible values
+     * enclosed by this range is more than the given value.
+     *
+     * @param value the value to compare with
+     * @return true if wider than the given value
+     */
+    public boolean isWiderThan(int value) {
```

Comment:
Will do. This function was intended for comparison with the MAX_VALUE (which is 10 currently). As you are suggesting we decouple this class with the checker, it makes sense to be `long` instead of `int`.